packages/gatsby-theme/src/components/Share/ShareWhatsapp.js

```diff
@@ -0,0 +1,32 @@
+import PropTypes from "prop-types"
+import { WhatsappShareButton } from "react-share"
+import Icon from "../Icon"
+
+const ShareWhatsapp = ({ title, to, separator, children, ...props }) => {
+  return (
+    <WhatsappShareButton
+      url={to}
+      title={title}
+      separator={separator}
+      resetButtonStyle={false}
+      {...props}
+    >
+      <Icon icon='whatsapp' />
+      {children}
+    </WhatsappShareButton>
```

Comment:
Same refactoring of using a single component with a Hashmap to resolve the specific component from the "react-share" library